components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/observer/ElasticMediationFlowObserver.java

```diff
@@ -118,7 +118,7 @@ public ElasticMediationFlowObserver() {
                 log.debug("Transport Client is built.");
             }
 
-            client.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
+            client.addTransportAddress(new TransportAddress(InetAddress.getByName(host), port));
```

Comment:
Would this change be backword compatible with earlier versions of elastic stack?

Reply:
The elastic search API has been changed from version 5 and 6. The InetSocketTransportAddress class has been removed from version 6 onwards. To overcome this we can replace InetSocketTransportAddress with TransportAddress class. Though this will not be backward compatible with older versions.